parquet-hadoop/src/main/java/org/apache/parquet/format/converter/ParquetMetadataConverter.java

```diff
@@ -1226,14 +1239,31 @@ public ParquetMetadata readParquetMetadata(InputStream from) throws IOException
   static FileMetaData filterFileMetaDataByMidpoint(FileMetaData metaData, RangeMetadataFilter filter) {
     List<RowGroup> rowGroups = metaData.getRow_groups();
     List<RowGroup> newRowGroups = new ArrayList<RowGroup>();
+    long preStartIndex = 0;
+    long preCompressedSize = 0;
+    boolean firstColumnWithMetadata = true;
+    if (rowGroups != null && rowGroups.size() > 0) {
+      firstColumnWithMetadata = rowGroups.get(0).getColumns().get(0).isSetMeta_data();
+    }
     for (RowGroup rowGroup : rowGroups) {
       long totalSize = 0;
       long startIndex;
+      ColumnChunk columnChunk = rowGroup.getColumns().get(0);
+      if (firstColumnWithMetadata) {
+        startIndex = getOffset(columnChunk);
+      } else {
+        assert rowGroup.isSetFile_offset();
+        assert rowGroup.isSetTotal_compressed_size();
 
-      if (rowGroup.isSetFile_offset()) {
+        //the file_offset of first block always holds the truth, while other blocks don't :
```

Comment:
I think this is no longer true with the issue we found in parquet-cpp.

Reply:
@sunchao Thanks for your information.May be the first rowgroup check is enough for this situation(firstFileOffset==4)? I will submit a new commit.